packages/frameworks-qwik/src/index.ts

```diff
@@ -0,0 +1,382 @@
+ *       <input type="hidden" name="providerId" value="github" />
+ *       <input type="hidden" name="options.callbackUrl" value="http://qwik-auth-example.com/dashboard" />
+ *       <button>Sign In</button>
+ *     </Form>
+ *   );
+ * });
+ *
+ * Sign In via code
+ *
+ * import { component$ } from '@builder.io/qwik';
+ * import { useAuthSignIn } from '~/routes/plugin@auth';
+ *
+ * export default component$(() => {
+ *   const signIn = useAuthSignIn();
+ *   return (
+ *     <button onClick$={() => signIn.submit({ providerId: 'github', options: { callbackUrl: 'http://qwik-auth-example.com/dashboard' } })}>Sign In</button>
+ *   );
+ * });
+ *
+ * Sign out via form
+ *
+ * import { component$ } from '@builder.io/qwik';
+ * import { Form } from '@builder.io/qwik-city';
+ * import { useAuthSignOut } from '~/routes/plugin@auth';
+ *
+ * export default component$(() => {
+ *   const signOut = useAuthSignOut();
+ *   return (
+ *     <Form action={signOut}>
+ *       <input type="hidden" name="callbackUrl" value="/signedout" />
+ *       <button>Sign Out</button>
+ *     </Form>
+ *   );
+ * });
+ *
+ * Sign out via code
+ *
+ * import { component$ } from '@builder.io/qwik';
+ * import { useAuthSignOut } from '~/routes/plugin@auth';
+ *
+ * export default component$(() => {
+ *   const signOut = useAuthSignOut();
+ *   return <button onClick$={() => signOut.submit({ callbackUrl: '/signedout' })}>Sign Out</button>;
+ * });
+ *
+ * ## Managing the session
+ *
+ * import { component$ } from '@builder.io/qwik';
+ * import { useAuthSession } from '~/routes/plugin@auth';
+ *
+ * export default component$(() => {
+ *   const session = useAuthSession();
+ *   return <p>{session.value?.user?.email}</p>;
+ * });
+ *
+ * ## Authorization
+ *
+ * Session data can be accessed via the route event.sharedMap.
+ * So a route can be protected and redirect using something like this located in a layout.tsx or page index.tsx:
+ *
+ * export const onRequest: RequestHandler = (event) => {
+ *   const session = event.sharedMap.get("session")
+ *   if (!session || new Date(session.expires) < new Date()) {
+ *     throw event.redirect(302, `/api/auth/signin?callbackUrl=${event.url.pathname}`);
+ *   }
+ * };
+ *
+ * ```
+ *
+ * @module @auth/qwik
+ */
+
+import { Auth, skipCSRFCheck } from "@auth/core"
+import { AuthAction, Session } from "@auth/core/types"
+import { implicit$FirstArg, type QRL } from "@builder.io/qwik"
+import {
+  globalAction$,
+  routeLoader$,
+  z,
+  zod$,
+  type RequestEventCommon,
+} from "@builder.io/qwik-city"
+import { EnvGetter } from "@builder.io/qwik-city/middleware/request-handler"
+import { isServer } from "@builder.io/qwik/build"
+import { parseString, splitCookiesString } from "set-cookie-parser"
+import { GetSessionResult, QwikAuthConfig, qwikAuthActions } from "./types.js"
+
+export function qwikAuthQrl(
+  authOptions: QRL<(ev: RequestEventCommon) => QwikAuthConfig>
+) {
+  const useAuthSignIn = globalAction$(
```

Comment:
IMO for simplicity, let's use `useSignIn`, `useSignOut`, and `useSession` as our exports